sonic-net/sonic-buildimage
# [System ready] Extend sysmonitor functionality to wait for host daemons

#### Why I did it
To extend sysmonitor to report services states according to HLD: https://github.com/sonic-net/SONiC/pull/1363

#### How I did it
* Add timeout monitor task to sysmonitor to declare system ready state forcibly when timout happened.
* sysmonitor now has a state machine, so the status now doesn't get back to "Not OK" after it was declared as "OK"
* A new "irrel_for_sysready" configuration was added to "FEATURE" table to handle services for which sysmonitor shouldn't wait for.
* Add "wait_services" field to platform health config file to explicitly specify services sysmonitor has to wait for.

#### How to verify it

* Start NOS on the box
* Make sure it defines system ready status
* Make sure `sysreadyshow` shows all services as OK 

#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes

[Device metadata config](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md#device-metadata)

#### A picture of a cute animal (it is my cat Finn)
![PXL_20230413_140941610 PORTRAIT](https://user-images.githubusercontent.com/35844154/231862413-cb630c8f-895f-4051-85e5-86802f9ccf40.jpg)
